libbeat/processors/add_locale/add_locale.go

```diff
@@ -0,0 +1,47 @@
+package actions
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/pkg/errors"
+)
+
+type addLocale struct {
+	timezone string
+}
+
+func init() {
+	processors.RegisterPlugin("add_locale", newAddLocale)
+}
+
+func newAddLocale(c common.Config) (processors.Processor, error) {
+	config := struct {
+		TimeZone string `config:"timezone"`
```

Comment:
I think idea is to report the local timezone. It probably should be configurable at all. Just detect the local TZ and report it in the event.

Once you have the timezone, you can use Logstash to do further transformations on the data (like interpreting syslog dates that are missing the timezone) or converting `@timestamp` to the machine's local timezone.

A more specific use case is analyzing logon events, the timezone can be used to determine if the event occurred during normal working hours.